app/app/settings.py

```diff
@@ -74,12 +74,20 @@
     'django.middleware.common.CommonMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.auth.middleware.RemoteUserMiddleware',  # added for custom SSO 
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
     'social_django.middleware.SocialAuthExceptionMiddleware',
     'applicationinsights.django.ApplicationInsightsMiddleware',
+    
+    'custom_auth.CustomSSOMiddleware' # added for custom SSO 
 ]
 
+# added for custom SSO 
+CREATE_USER='TRUE'
+IS_SUPER=1 # 1:True | 0: False 
```

Comment:
If we set this globally it means that every user who authenticates via a header will be an admin. Superuser status should be set on a per-request basis if a specific header such as `X-USER-IS-ADMIN` is passed. This will allow Doccano to respect any group memberships that the authentication proxy passes through.

Reply:
Currently, ```IS_SUPER``` does not impact the login process for an existing user who is not an admin. Only the user who were created or have their ```user.is_superuser``` property set to ```true``` are logged in as an admin. 

Let me know if that make sense. I can add the extra header but wanted to make sure.

Reply:
Currently every user who logs in via the header will be an admin, right? Let's make that configurable on a per-request/per-login basis via the auth proxy passing a specific/configurable header.

Reply:
No. Only the user with the header ```HTTP_X_CUSTOM_USER_NAME``` and ```user.is_superuser``` set to ```true``` will be logged in an admin. Now, if the user does not exist in the database then the user will be created on the fly and ```IS_SUPER``` flag will determine whether they are an admin or not. Also, if ```IS_SUPER``` is ```false``` they won't be logged in as an admin.

Reply:
The determination of whether a user is admin or not should happen on a per-request basis so that the authentication proxy can make someone an admin via a header. The idea here is to enable the following workflow:

1. UserA logs into the authentication proxy and has admin access to Doccano. The authentication proxy passes `X-USER: UserA`, `X-Groups: Admin`. Doccano picks this up and logs the user in as an admin.

2. UserB logs into the authentication proxy and does not have admin access to Doccano. The authentication proxy passes `X-USER: UserB`. Doccano picks this up and logs the user in as a non-admin.

If we set the is-admin flag via the environment, then both UserA and UserB would be logged in as admin since `X-Groups` is ignored.

Reply:
I have updated the code:

You can test the cases:

1.  UserA will be granded full Admin level priviliges.

- ```curl -I "http://localhost:8000" -H "X-AUTH-PROXY-USER: UserA"  -H "X-AUTH-PROXY-GROUPS: Admin" ```    

2.  UserA existing Admin level priviliges will be revoked (since```X-AUTH-PROXY-GROUPS:``` header no longer pass ```Admin``` )

- ```curl -I "http://localhost:8000" -H "X-AUTH-PROXY-USER: UserA"  -H "X-AUTH-PROXY-GROUPS: Guest" ```    

3. ```IS_SUPER``` is set to ```True``` in ```settings.py```. ```X-AUTH-PROXY-GROUPS:```  will be ignored and UserA will be granded full Admin level priviliges.

- ```curl -I "http://localhost:8000" -H "X-AUTH-PROXY-USER: UserA"  -H "X-AUTH-PROXY-GROUPS: Guest" ```   

4. If  no ```X-AUTH-PROXY-GROUPS:``` is passed then UserB will be ganred non Admin level priviliges.

- ```curl -I "http://localhost:8000" -H "X-AUTH-PROXY-USER: UserB"  ```